scripts/taco_to_burrito.py

```diff
@@ -90,6 +101,9 @@ def parse_marker_category(marker_category_node, base=""):
     else:
         limited_attribs["height_offset"] = 0
 
+    if "texture" in attribs:
```

Comment:
Does `"texture"` need to be added to the `allowed_keys` above? I feel like this would cause an error if the texture attribute existed in the marker category

Reply:
Technically speaking yes, although "cause an error" in this case only means "print an error message". I've added all previously unknown attributes that I've encountered while converting Teh's Trails.

I'm not really sure why most of those checks are done, it seems most 'required' attributes are never used anyway.

Reply:
I've indeed used Teh's Trails from the link you posted. Although some filenames and references are broken ([PR opened](https://github.com/xrandox/TehsTrails/pull/20)). The 'only' remaining issue is that multiple identical Trails (different texture/colors) are defined, which overlap and clip through one another. I think in TacO you can just select / put checkboxes on the Trails you want to see. For me personally I removed all but one of the Trail definitions.

Reply:
Good point about the required attributes not being used. The goal was reverse engineering the spec not dictating it. So if it looked like an attribute was required in the spec it would go into "required" until a counterexample was found, then it would be moved to allowed. If an error was printed about a key not being allowed that just meant that some attribute had not yet had human consideration for how it should be converted.

I think you are correct about TacO's checkboxes BlishHUD definitely does that and there is some conversation on it in #11 and #3 .